Add llvm-project archive issues for Chromium bug tracker

smithp35 opened the pull request:
The Chromium bug tracker is in an archived state. The Security Response Group has preemptively created llvm-project GitHub issues with PDF copies of the Chromium issues should the repository become inaccessible.

* Add URLs for redirects from https://bugs.chromium.org/p/llvm/issues/detail?id=X to https://issuetracker.google.com/issues/y
* Add URLs to llvm-project archive issues.
* Add an explanation of archive use.

smithp35 committed:
Add llvm-project archive issues for Chromium bug tracker

The Chromium bug tracker is in an archived state. The Security
Response Group has preemptively created llvm-project GitHub
issues with PDF copies of the Chromium issues should the
repository become inaccessible.

* Replace URLs in format:
  https://bugs.chromium.org/p/llvm/issues/detail?id=X with
  their redirect https://issuetracker.google.com/issues/y
* Add URLs to llvm-project archive issues.
* Add an explanation of archive use.

smithp35 commented:
Added as many from the security group as I could autocomplete from GitHub reviewers. I don't think this needs everyone to review as this isn't a policy change.

kbeyls commented on llvm/docs/SecurityTransparencyReports.rst:
```diff
@@ -2,7 +2,15 @@
+This page lists the yearly LLVM Security Response group transparency reports.
+
+The LLVM Security Response group started out as the LLVM security group, previous
+year's transparency reports keep the original year.
```
maybe s/keep the original year/keep the original name/?

smithp35 replied:
Thanks for spotting. Done

kbeyls commented on llvm/docs/SecurityTransparencyReports.rst:
```diff
@@ -2,7 +2,15 @@
+year's transparency reports keep the original year.
+
+Initially the Chromium issue tracker was used to record issues. This component
+has been archived and is read-only. A llvm-project issue with the SecurityArchive
```
Maybe slightly rephrase the last sentence to: "A github llvm/llvm-project issue has been created for each issue in the Chromium issue tracker. All such issues contain an attached PDF with the content of the Chromium issue. All such issues are labeled with the `SecurityArchive <https://github.com/llvm/llvm-project/issues?q=label%3ASecurityArchive>`_ label.

smithp35 replied:
I've reworded to get close to the text above.

kbeyls replied:
Looks good, thanks!

kbeyls commented on llvm/docs/SecurityTransparencyReports.rst:
```diff
@@ -29,8 +37,11 @@ In 2021, the security group received 13 issue reports that were made publicly
 
-* https://bugs.chromium.org/p/llvm/issues/detail?id=5
-* https://bugs.chromium.org/p/llvm/issues/detail?id=11
+* https://issuetracker.google.com/issues/42410043 archive:
```
I think it might be good to keep the original bugs.chromium.org urls, so that mapping from original chromium id remains possible. But this is only a weakly held opinion on my part...

smithp35 replied:
I'll have a think about this. For the newly created bugs I chose the redirected URL as that's what people will see when they click on the link. Which makes it easier to cross check.

Happy to change back if there's a consensus, I think I'll need to update the links in the llvm-project issues too.

smithp35 replied:
I've updated all the Github issues I've created to contain the original URLs. Current form using https://github.com/llvm/llvm-project/issues/131849 as an example:
```
This is an archival issue from the Security Group's Chromium bug-tracker which has now been archived. Capturing the contents of the Chromium issue in a PDF in case link.

https://issuetracker.google.com/u/1/issues/42410036 (Previous URL https://bugs.chromium.org/p/llvm/issues/detail?id=43) Vulnerable Version of xml2js Used in MLIR Manifest File

Published in https://llvm.org/docs/SecurityTransparencyReports.html#id3

[Vulnerable Version of xml2js Used in MLIR Manifest File [42410036] - Issue Tracker.pdf](https://github.com/user-attachments/files/19323328/Vulnerable.Version.of.xml2js.Used.in.MLIR.Manifest.File.42410036.-.Issue.Tracker.pdf)
```

smithp35 replied:
To resolve the original vs redirect URL I've found a way to include both for each issue. This does have the advantage that it is easier to check all the URLs are correct and refer to the same issue. I've double checked all the URLs with a redirect against the details in the archive.

serge-sans-paille commented on llvm/docs/SecurityTransparencyReports.rst:
```diff
@@ -2,7 +2,15 @@
+Initially the Chromium issue tracker was used to record issues. This component
+has been archived and is read-only. A llvm-project issue with the SecurityArchive
+label has been created with a further backup PDF copy in a llvm-project
+repository.
```
It took me time to understand that this actually means "Each of the original  issue from the chromium issue tracker has been mirrored as an llvm-project issue that keeps a link to a further backup PDF copy stored as an issue attachment." (feel free to rephrase, but I bugged on "with a further backup PDF copy in a llvm-project
repository")

smithp35 committed:
Kristof's Review comments

* year should have been name.
* reworded part about archives.

I've kept the URLs the same for now.

smithp35 committed:
Merge branch 'llvm:main' into transparencyurls

tuliom commented on llvm/docs/SecurityTransparencyReports.rst:
```diff
@@ -2,7 +2,17 @@
 LLVM Security Group Transparency Reports
```
This is probably beyond the scope of this PR, but it looks like the name of the group needs to be updated in the title.

smithp35 commented:
I've rebased my change on top of the 2024 Transparency report. I'll raise issues for the new Chromium issues on that page and update.

smithp35 committed:
Rebase on 2024 Transparency update

I've added Github issues for all the new Chromium issue tracker
entries.

smithp35 committed:
Updated to use both original, redirect and archive URLs.

Original URL cited for consistency with previous transparency
report. Redirect URL cited in case that is ever turned off.
Archive URL cited in case Chromium data deleted for whatever
reason.

kbeyls commented on llvm/docs/SecurityTransparencyReports.rst:
```diff
@@ -2,7 +2,21 @@
+label.
+
+Each Chromium issue has 3 URLs, the first is the original URL recorded in
+previous transparency reports. the second is the redirect URL to the archive.
```
s/the/The/ ?

smithp35 replied:
Thanks for spotting, will fix.

kbeyls reviewed (approved):
Thank you for this work @smithp35 !
LGTM, (I just saw one extremely tiny typo).

smithp35 committed:
Fix typo . the -> . The
